paimon-core/src/main/java/org/apache/paimon/utils/SnapshotManager.java

```diff
@@ -883,6 +884,6 @@ public void commitEarliestHint(long snapshotId) throws IOException {
 
     private void commitHint(long snapshotId, String fileName, Path dir) throws IOException {
         Path hintFile = new Path(dir, fileName);
-        fileIO.overwriteFileUtf8(hintFile, String.valueOf(snapshotId));
+        FileIOUtils.overwriteFileUtf8WithRetry(fileIO, hintFile, String.valueOf(snapshotId), 3);
```

Comment:
Hard code here? Or make it configurable better？

Reply:
ok